packages/retro-application-extension/src/index.ts

```diff
@@ -30,9 +34,33 @@ const NOTEBOOK_FACTORY = 'Notebook';
 const EDITOR_FACTORY = 'Editor';
 
 /**
- * A regular expression to match path to notebooks and documents
+ * A regular expression to match path to notebooks, documents and consoles
  */
-const TREE_PATTERN = new RegExp('/(notebooks|edit)\\/?');
+const URL_PATTERN = new RegExp('/(notebooks|edit|consoles)\\/?');
+
+/**
+ * Open consoles in a new tab.
+ */
+const consoles: JupyterFrontEndPlugin<void> = {
+  id: '@jupyterlite/retro-application-extension:consoles',
+  requires: [IConsoleTracker],
+  autoStart: true,
+  activate: (app: JupyterFrontEnd, tracker: IConsoleTracker) => {
+    const baseUrl = PageConfig.getBaseUrl();
+    tracker.widgetAdded.connect(async (send, console) => {
+      const widget = find(app.shell.widgets('main'), w => w.id === console.id);
+      if (widget) {
+        // bail if the console is already added to the main area
+        return;
+      }
+      const path = console.sessionContext.path;
+      window.open(`${baseUrl}retro/consoles?path=${path}`, '_blank');
```

Comment:
Since there will be only one kernel running on the page at a time (at least for now), we might not even need to specify the path here.

Reply:
For now let's keep it for consistency.